rfcs/054-prismic-api/README.md

```diff
@@ -0,0 +1,110 @@
+- The new API service should be written in typescript, following patterns set by the [concepts API](../050-concepts-api/README.md) for filtering, pagination, error handling, etc.
+- We shouldn't create new IDs for exhibitions, events, stories, etc. There aren't any situations where we should need to merge content from multiple sources, so we should use the document IDs directly from prismic.
+- The elasticsearch index mapping should represent the contract between the pipeline and the API. The API shouldn't need to know anything about the structure of the data in prismic.
+
+## Proposed endpoints
+
+### `/articles`
+
+Filter for type eg stories, webcomics
+
+#### Different kinds of thing (are these 'labels'?)
+
+- story <https://wellcomecollection.org/articles/Y9vktREAACYA1v2U> at /articles
+- webcomic <https://wellcomecollection.org/articles/Y86rshEAACcAI4g4> at /articles
+- book <https://wellcomecollection.org/books/Y7QWrREAAHC43oH0> at /books
+- series <https://wellcomecollection.org/series/WfpEBCoAACrdd43y> composed of articles
+- in pictures <https://wellcomecollection.org/articles/Y-oAIhAAACAAix32> at /articles
+
+#### Filters
+
+- date
+- contributor
+- series
+- contentType
+
+#### Sort
+
+- date
+- relevance
+
+### `/exhibitions`
+
+#### Filter
+
+- date
+- status
+- location
+- contributor?
+- format (ie permanent, exhibition, installation)
+- season
+
+#### Sort
+
+- date
+- relevance
+
+### `/events`
+
+#### Filter
+
+- date
+- accessibility status (is it audio described, signed, etc)
+- location
+- format (eg discussion)
+- event-series
+- season
+- audience (eg schools)
+- contributor
+- isOnline?
+- availableOnline?
+
+#### Sort
+
+- date
+- relevance
+
+### `/books`
+
+?????????
+
+## Open questions
+
+- How do we represent functional content?
+- Should we include the full prismic API response at eg `/articles/{id}`?
+    - should we be able to list all of the docs of a given type at once? what should that url look like?
+- If we want to avoid maintaining multiple versions of the same code, should this work be done as an extension to the concepts API?
+- Works exist at [api.wellcomecollection.org/catalogue/v2/works](api.wellcomecollection.org/catalogue/v2/works), concepts at [/catalogue/v2/concepts](api.wellcomecollection.org/catalogue/v2/concepts), images at [/catalogue/v2/images](api.wellcomecollection.org/catalogue/v2/images).  
+Stories etc aren't part of the catalogue - should the URL be different?
+- what are the top level types? is it articles? stories? series?
+- where do books go? they appear on the stories landing page, so should they appear in stories search results? or are they at a separate endpoint?
+- what is a serial? what is a series?
+- is 'In pictures' the same as 'Image gallery'?
+- Should we be able to filter by contributor role? is that something for the first pass or something we address later? which contributor roles are there? yes for stories? no for exhibitions and events?
```

Comment:
@jtweed one for you. Considering authors & artists (which contain sub-categories such as photographers, illustrators, etc.), maybe there are others. 

We think it might make more sense for stories, which will often have more than one result per author, versus exh./events.

Reply:
If we allow filtering by contribution role in Prismic, do we also allow it for works?

Reply:
We could also figure out later with data if people are trying to search for specific artists and take it from there?

Reply:
@alexwlchan made the point yesterday that filtering exhibitions and events by contributor is necessary if we want to include those results on a contributor's concept page. IMO that's more than enough reason to include it as a filter!

We should still clarify whether we want to be able to filter by specific contributor role though

Reply:
I absolutely agree, I meant the split by roles, sorry! 

So one filter for now and eventually, is this a nice one to have specific filters for roles?
I'm thinking especially of stories, where the author and the artist are different and people might want to know they're looking for the photographer/illustrator, etc.